### PR TITLE
issue #153 - remove all references to the "virtual resource" function

### PR DESCRIPTION
**Pull request by lmsurpre**
1. Removed FHIRVirtualResourceTypeException; now just throw FHIRHttpException
2. Changed response status from 400 to 404 when client performs request on a resourceType endpoint that isn't a valid resource type
3. Unrelated cleanup: Updated FHIRUtil buildOperationOutcome methods to
accept elements and not just the ValueSet (simplifies usage across the
board)

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-config/src/test/java/com/ibm/fhir/config/test/ConfigurationServiceTest.java [hunk 1]
```diff
@@ -33,11 +33,13 @@ public void testLoadConfiguration() throws Exception {
         assertNotNull(pg);
                 
         // Validate retrieval of an array of strings.
-        Object[] allowableCustomTypes = pg.getArrayProperty("fhirServer/virtualResources/allowableResourceTypes");
-        assertNotNull(allowableCustomTypes);
-        assertEquals(2, allowableCustomTypes.length);
-        assertEquals("WeatherDetail", (String) allowableCustomTypes[0]);
-        assertEquals("XYZStudy", (String) allowableCustomTypes[1]);
+        //Object[] allowableCustomTypes = pg.getArrayProperty("fhirServer/virtualResources/allowableResourceTypes");
```
Comment (prb112) [inline fhir-config/src/test/java/com/ibm/fhir/config/test/ConfigurationServiceTest.java]: remove this line?

File: fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java [hunk 2]
```diff
@@ -797,36 +797,14 @@ public static boolean hasTag(Resource resource, Coding tag) {
...
     /**
-     * Returns the resource type (as a String) of the specified resource. For a virtual resource, this will be the
-     * actual virtual resource type (not Basic).
+     * Returns the resource type (as a String) of the specified resource. 
      * 
      * @param resource
      *            the resource
      * @return the name of the resource type associated with the resource
      */
     public static String getResourceTypeName(Resource resource) {
```
Comment (prb112) [inline fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java]: should we deprecate this method?  and point to something in ModelSupport?

File: fhir-server/liberty-config-tenants/config/tenant2/fhir-server-config.json [hunk 3]
```diff
@@ -1,14 +1,8 @@
 {
-    "__comment":"FHIR Server configuration for mythical tenant id 'tenant1'",
+    "__comment":"FHIR Server configuration for mythical tenant id 'tenant2'",
```
Comment (prb112) [inline fhir-server/liberty-config-tenants/config/tenant2/fhir-server-config.json]: Good catch

File: fhir-config/src/test/java/com/ibm/fhir/config/test/ConfigurationServiceTest.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-config/src/test/java/com/ibm/fhir/config/test/ConfigurationServiceTest.java]: for sure
```suggestion
```

File: fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java]: sorry, missed this one before.  we discussed on our slack channel a bit and John also thought it should be deprecated, but I think his suggestion was just to use instance.getClass().getSimpleName() everywhere.  I sort of liked the idea of having the helper wrap this (even though its just one line), but I don't care much where it lives.

Comment (prb112) [reply fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java]: No worries, I didn't think this was a big deal.  I like the helper method approach. FHIRUtil seems fine for now :)

**Pull request merged**